src/main/java/me/wiefferink/bukkitdo/Do.java

```diff
@@ -159,7 +161,7 @@ public void run() {
 	 * @param period   Time between task runs
 	 * @return BukkitTask which can be used to cancel the operation
 	 */
-	public static BukkitTask asyncTimer(long period, RunResult<Boolean> runnable) {
+	public static BukkitTask asyncTimer(long period, Supplier<Boolean> runnable) {
```

Comment:
Suggest that all uses of Supplier should probably instead be Callable.
But I could be an idiot here, Callable would enable throwing exceptions.

Reply:
i converted your functional interfaces with the JDK's internal interfaces so, there is no any difference between yours and JDKs. they were redunant.